utils/utils.py

```diff
@@ -219,7 +202,7 @@ def read_csv(csv_file_name, inference=False):
         return sorted(list_values, key=lambda k: k['dataset'])
 
 
-def get_device_ids(number_requested):
+def get_device_ids(number_requested): #FIXME if some memory is used on a GPU before call to this function, the GPU will be excluded.
```

Comment:
`if some memory is used on a GPU before call to this function, the GPU will be excluded`... That's exactly what this function does. Since we don't know in advance how much memory is needed for the process we want to push to the GPU, we exclude those with data on it. There might be a better way to solve this, tough. Any suggestions?

Reply:
Actually, my note is unclear. What I meant is:
Sometimes, a bit of GPU memory is used by a python script (for what reason, I don't know...) but you still want to use that GPU. Therefore I modified code to include GPU even if, say, 10% of it's memory is used. I also added a threshold for the %usage for the same reason. 

The function was fine to begin with, I just raised the threshold (harcoded) to include partly used GPUs.